apis/caller/AxiosCaller.ts

```diff
@@ -0,0 +1,60 @@
+import axios, {
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosRequestHeaders,
+  InternalAxiosRequestConfig,
+} from "axios";
+import { AUTH_NEEDED_URI, API_URL, DEFAULT_TIMEOUT } from "../constants";
+import { ICaller } from "./types";
+
+export class AxiosCaller implements ICaller<AxiosRequestConfig> {
+  axios: AxiosInstance;
+  constructor() {
+    this.axios = axios.create({
+      baseURL: API_URL,
+      timeout: DEFAULT_TIMEOUT,
+    });
+
+    this.axios.interceptors.request.use((config) => {
+      const token = localStorage.getItem("token");
+      const authRequired = AUTH_NEEDED_URI.some((pattern) => {
+        return config.url?.match(pattern);
+      });
+      if (!token && authRequired) throw new Error("Unauthorized");
```

Comment:
나중에 에러처리를 고도화시킬 때 수정되어야하니까 
```suggestion
// TODO: ERORR
// 에러객체를 만들때 변경되어야함
 if (!token && authRequired) throw new Error("Unauthorized");
```
이렇게 추가 되면 좋을 것 같습니다.

Reply:
#7 이슈로 열어 놓았습니다. 병합은 여기에서 해두고, 추후 작업에서는 분리하는 작업을 할 수 있도록 하겠습니다.